File: repo/atlas-foundry/server/deploy.go
package server

import (
	"context"
	"math/big"
)

type DeploymentParams struct {
	Name          string
	Symbol        string
	Supply        *big.Int
	Owner         string
	WalletClient  interface{}
}

type DeploymentResult struct {
	ContractAddress string
	TxHash          string
}

func DeployERC20(ctx context.Context, params *DeploymentParams) (*DeploymentResult, error) {
	return &DeploymentResult{
		ContractAddress: "0x...",
		TxHash:          "0x...",
	}, nil
}

File: repo/atlas-foundry/server/deploy.ts
import { parseUnits, encodeFunctionData, erc20Abi } from 'viem';
import type { WalletClient } from 'viem';

export interface ERC20DeploymentParams {
  name: string;
  symbol: string;
  supply: bigint;
  owner: `0x${string}`;
  walletClient: WalletClient;
}

export interface ERC20DeploymentResult {
  contractAddress: `0x${string}`;
  txHash: `0x${string}`;
}

export async function deployERC20(
  params: ERC20DeploymentParams
): Promise<ERC20DeploymentResult> {
  const [account] = await params.walletClient.getAddresses();
  
  const hash = await params.walletClient.sendTransaction({
    account,
    to: undefined,
    data: encodeContractCreation(params.name, params.symbol, params.supply),
  });

  const receipt = await params.walletClient.waitForTransactionReceipt({ hash });
  
  if (!receipt.contractAddress) {
    throw new Error('Contract deployment failed');
  }

  return {
    contractAddress: receipt.contractAddress,
    txHash: receipt.transactionHash,
  };
}

function encodeContractCreation(
  name: string,
  symbol: string,
  supply: bigint
): `0x${string}` {
  return '0x608060405234801561001057600080fd5b50...' as `0x${string}`;
}






